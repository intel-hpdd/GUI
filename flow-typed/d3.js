// @flow

declare module d3 {
  declare module.exports: {
    max(xs:number[]):number;
    extent(xs:number[]):number[];
    scale:Object;
    time:Object;
  }
}
