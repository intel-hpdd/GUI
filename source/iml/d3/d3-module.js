// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import d3 from 'd3';

export default angular.module('d3', []).value('d3', d3).name;
