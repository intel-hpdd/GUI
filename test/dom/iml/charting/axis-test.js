// @flow

import { render } from "inferno";

import Axis from "../../../../source/iml/charting/axis.js";
import d3 from "d3";

describe("axis", () => {
  let svg, chartingGroup;

  beforeEach(() => {
    svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
    chartingGroup = document.createElementNS("http://www.w3.org/2000/svg", "g");
    svg.appendChild(chartingGroup);
  });

  it("should render x as expected", () => {
    render(
      <Axis
        type="x"
        xScale={d3.scale
          .linear()
          .domain([0, 100])
          .range([0, 200])}
        dimensions={{ usableWidth: 200, usableHeight: 200 }}
        chartingGroup={d3.select(chartingGroup).datum([0, 100])}
      />,
      svg
    );

    expect(svg).toMatchSnapshot();
  });

  it("should render y as expected", () => {
    render(
      <Axis
        type="y"
        yScale={d3.scale
          .linear()
          .domain([0, 100])
          .range([0, 200])}
        dimensions={{ usableWidth: 200, usableHeight: 200 }}
        chartingGroup={d3.select(chartingGroup).datum([0, 100])}
      />,
      svg
    );

    expect(svg).toMatchSnapshot();

    render(null, svg);
  });

  it("should throw when props are missing", () => {
    expect.assertions(2);

    expect(() =>
      render(
        <Axis
          type="y"
          yScale={d3.scale
            .linear()
            .domain([0, 100])
            .range([0, 200])}
          chartingGroup={d3.select(chartingGroup).datum([0, 100])}
        />,
        svg
      )
    ).toThrow();

    expect(() =>
      render(
        <Axis
          type="y"
          yScale={d3.scale
            .linear()
            .domain([0, 100])
            .range([0, 200])}
          dimensions={{ usableWidth: 200, usableHeight: 200 }}
        />,
        svg
      )
    ).toThrow();
  });

  it("should skip cleanup", () => {
    render(
      <Axis
        type="y"
        yScale={d3.scale
          .linear()
          .domain([0, 100])
          .range([0, 200])}
        dimensions={{ usableWidth: 200, usableHeight: 200 }}
        chartingGroup={d3.select(chartingGroup).datum([0, 100])}
      />,
      svg
    );

    render(
      <Axis
        type="y"
        yScale={d3.scale
          .linear()
          .domain([0, 100])
          .range([0, 200])}
        dimensions={{ usableWidth: 200, usableHeight: 200 }}
      />,
      svg
    );

    render(null, svg);
  });
});
