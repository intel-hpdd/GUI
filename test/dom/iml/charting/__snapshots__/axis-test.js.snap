// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axis should render x as expected 1`] = `
<svg>
  <g>
    <g
      class="axis x-axis"
      transform="translate(0,180)"
    >
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(20,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          10
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(40,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          20
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(60,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          30
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(80,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          40
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(100,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          50
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(120,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          60
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(140,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          70
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(160,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          80
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(180,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          90
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(200,0)"
      >
        <line
          x2="0"
          y2="6"
        />
        <text
          dy=".71em"
          style="text-anchor: middle;"
          x="0"
          y="9"
        >
          100
        </text>
      </g>
      <path
        class="domain"
        d="M0,6V0H200V6"
      />
    </g>
  </g>
  
</svg>
`;

exports[`axis should render y as expected 1`] = `
<svg>
  <g>
    <g
      class="axis y-axis"
    >
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,0)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,20)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          10
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,40)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          20
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,60)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          30
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,80)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          40
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,100)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          50
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,120)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          60
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,140)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          70
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,160)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          80
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,180)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          90
        </text>
      </g>
      <g
        class="tick"
        style="opacity: 1;"
        transform="translate(0,200)"
      >
        <line
          x2="-6"
          y2="0"
        />
        <text
          dy=".32em"
          style="text-anchor: end;"
          x="-9"
          y="0"
        >
          100
        </text>
      </g>
      <path
        class="domain"
        d="M-6,0H0V200H-6"
      />
    </g>
  </g>
  
</svg>
`;
